routes/index.js

```diff
@@ -14,12 +16,36 @@ router.get("/login", function (req, res, next) {
   res.render("login");
 });
 
-router.get('/signup/phone', function (req, res, next) {
-  res.render('signup-phone', { title: '배민 - 전화번호 인증'})
+router.get("/signup/phone", function (req, res, next) {
+  res.render("signup-phone", { title: "배민 - 전화번호 인증" });
 });
 
 router.get("/signup/detail", function (req, res, next) {
-  res.render("signup-detail");
+  const hasSignupError = req.session?.signupError;
+  if (!hasSignupError) {
+    res.render("signup-detail", { error: null });
+    return;
+  }
+
+  req.session.signupError = null;
+  res.render("signup-detail", { error: hasSignupError });
 });
 
+router.post("/signup", async function (req, res, next) {
+  const { email, password, nickname } = req.body;
+
+  const hasAlreadyUserEmail = await db.find({ email });
+
+  if (hasAlreadyUserEmail.length) {
```

Comment:
```suggestion
  const foundUsers = await db.find({ email });

  if (foundUsers.length) {
```

반화되는 결과가 list of object이니 boolean을 의미하는 has로 시작하는 변수는 맞지 않는것 같습니다.

Reply:
넵! 좋은 의견 감사합니다.

다만 중복체크로 인해 db에 동일한 이메일을 가진 유저가 존재하지 않으니

`foundUsers` => `foundUser`로 변경하여 커밋하겠습니다~

Reply:
음... 코드 상의 query에 의해서는 zero or one 결과 레코드가 나오는게 맞긴 하지만 find의 결과 타입은 array of object가 맞으니 users가 맞다고 생각하지만... 이러한 문제를 해결하기 위해서는 if 부분을 고쳐야 할 것 같습니다.

``` javascript
if (foundUser)
```

이렇게 하는거죠. javascript에서는 빈 배열을 false와 동일하게 취급하니 말입니다.

Reply:
으아닛! 빈배열이 true로 처리되네요! 이 말은 철회합니다 ㅠ